### PR TITLE
fix: alter serial column

### PR DESCRIPTION
**Pull request by a631807682**
- [x] Do only one thing
- [x] Non breaking API changes
- [ ] Tested

### What did this pull request do?
`AlterColumn` throw error when type sercial
Do I need to create a test on gorm?

https://github.com/go-gorm/gorm/issues/5047

### CODE REVIEW

File: migrator.go [hunk 1]
```diff
@@ -475,3 +494,74 @@ func (m Migrator) CurrentSchema(stmt *gorm.Statement, table string) (interface{}
...
+	if err := tx.Exec("ALTER TABLE ? ALTER COLUMN ? TYPE ?", m.CurrentTable(stmt), clause.Column{Name: field.DBName}, fileType).Error; err != nil {
+		return err
+	}
+
+	if err := tx.Exec("ALTER TABLE ? ALTER COLUMN ? DROP DEFAULT",
+		m.CurrentTable(stmt), clause.Expr{SQL: field.DBName}).Error; err != nil {
+		return err
+	}
+
+	// is it necessary to delete sequence ?
```
Comment (a631807682) [inline migrator.go:565]: I'm not sure if I need to delete the sequence

File: migrator.go [hunk 2]
```diff
@@ -249,8 +249,27 @@ func (m Migrator) AlterColumn(value interface{}, field string) error {
 			return m.DB.Connection(func(tx *gorm.DB) error {
 				fileType := clause.Expr{SQL: m.DataTypeOf(field)}
 				if fieldColumnType.DatabaseTypeName() != fileType.SQL {
-					if err := tx.Exec("ALTER TABLE ? ALTER COLUMN ? TYPE ?", m.CurrentTable(stmt), clause.Column{Name: field.DBName}, fileType).Error; err != nil {
-						return err
+					fileColumnAutoIncrement, _ := fieldColumnType.AutoIncrement()
```
Comment (jinzhu) [inline migrator.go:252]: `fileColumnAutoIncrement` -> `filedColumnAutoIncrement`?

File: migrator.go (hunk 1 above)
Comment (jinzhu) [reply migrator.go:565]: will it return error if it is still used?

**Comment by jinzhu** [comment]
In general look great, thank you for your PR.

**Pull request merged**